image_view/src/nodes/video_recorder.cpp

```diff
@@ -70,9 +71,14 @@ void callback(const sensor_msgs::ImageConstPtr& image_msg)
 
     }
 
-    if ((image_msg->header.stamp - g_last_wrote_time) < ros::Duration(1.0 / fps))
+    if ((image_msg->header.stamp - g_last_wrote_time) < ros::Duration((1.0 / fps)*(1.0-tolerance/100)))
     {
       // Skip to get video with correct fps
+      ROS_INFO_STREAM("Skipping frame "<< g_count <<": "
+                << image_msg->header.stamp - g_last_wrote_time << " sec is outside of the " 
+                << tolerance<< "\% tolerance");
```

Comment:
Since this is likely to happen often on some systems, I would prefer this be a `_THROTTLE` at minimum and a `_DEBUG` if you're OK with it instead of an `_INFO`. Terminal spam is not recommended.

Reply:
I understand the concern over terminal spam, so I'd be fine with setting this to `_DEBUG`.

I'm not sure what you mean by "throttle at minimum"? Do you mean I should change the debug print statement to say "_Throttle at minimum framerate_" or something like that?

Reply:
Sorry, no, I meant using one of the `_THROTTLE` macros so the output of this message is throttled to once per time _t_ (see http://wiki.ros.org/roscpp/Overview/Logging - the Throttle section).

Reply:
Oh nice! That's super handy. I didn't know the `_THROTTLE` type existed. Okay, so then what's a reasonable period? Once per second? Before, there was no notice of skipped frames at all. Do you think once every 2 seconds is fine?

Another thing: I'm not sure if it matters but I also changed the way frames are counted here (before, the skipped frames were not counted in `g_count`. But I felt it made sense for `g_count` to be "every frame that has ever arrived" rather than just "every frame that was saved". Thoughts on this?